File: Travel.WebAPI/Controllers/API/LocationController.cs
using System;
using System.Net;
using System.Web.Http;
using System.Web.Script.Serialization;
using System.Net.Http;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Travel.WebAPI.Controllers.API {

    public class LocationController : ApiController
    {


        private JavaScriptSerializer jss = new JavaScriptSerializer();
        // GET api/<controller>
        public string GetValues()
        {

            List<Models.LocationModel> Locations = AutoMapper.Mapper.Map<List<DAL.Location>, List<Models.LocationModel>>(Business.Location.SelectLocations());
            return JsonConvert.SerializeObject(Locations);

        }


        // GET api/location/Graph
        [Route("api/location/Graph")]
        public string GetGraphTable()
        {

            dynamic googleDataTable = new Bortosky.Google.Visualization.GoogleDataTable(Business.Location.SelectLocationsForMap());
            string Json = googleDataTable.GetJson();
            return Json;
        }

        // GET api/<controller>/5
        public string GetValue(int id)
        {

            if (id == 0)
            {
                dynamic googleDataTable = new Bortosky.Google.Visualization.GoogleDataTable(Business.Location.SelectLastKnownLocationForMap());
                string Json = googleDataTable.GetJson();

                return Json;
            }

            return "Not Configured";
        }

        // POST api/<controller>
        [Authorize()]
        public HttpResponseMessage PostValue([FromBody()]  DAL.Location Location)
        {

            try
            {
                if (!Business.Location.EntryExistsForToday(Location))
                    Business.Location.Insert(Location);

            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.Conflict, ex.Message);
            }

          return  Request.CreateResponse(HttpStatusCode.OK);

        }

        // DELETE: api/<controller>/5
        [Authorize()]
        public HttpResponseMessage Delete(int id)
        {
            try
            {
                Business.Location.Delete(id);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.Conflict, ex.Message);
            }

            return Request.CreateResponse(HttpStatusCode.OK);

        }

        // PUT api/<controller>/5
        [Authorize()]
        public HttpResponseMessage Put(int id, [FromBody] DAL.Location location)
        {
            try
            {
                Business.Location.Update(location, id);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.Conflict, ex.Message);
            }

            return Request.CreateResponse(HttpStatusCode.OK);
        }

        // GET api/location/lastknown
        [HttpGet()]
        [Route("api/location/lastknown")]
        public string LastKnown()
        {
            return Business.Location.SetLastKnownLocation();
        }
    }
}

File: Travel.Web/ServiceAPI/LocationController.vb
Imports System.Net
Imports System.Web.Http
Imports System.Web.Script.Serialization
Imports System.Net.Http

Public Class LocationController
    Inherits ApiController

    Private jss As New JavaScriptSerializer()

    ' GET api/<controller>
    Public Function GetValues() As String

        Dim googleDataTable = New Bortosky.Google.Visualization.GoogleDataTable(Business.Location.SelectLocationsForMap)
        Dim Json As String = googleDataTable.GetJson
        Return Json
    End Function

    ' GET api/<controller>/5
    Public Function GetValue(ByVal id As Integer) As String

        If id = 0 Then
            Dim googleDataTable = New Bortosky.Google.Visualization.GoogleDataTable(Business.Location.SelectLastKnownLocationForMap)
            Dim Json As String = googleDataTable.GetJson

            Return Json
        End If

        Return "Not Configured"
    End Function

    ' POST api/<controller>
    <Authorize(Roles:="Administrators")>
    Public Function PostValue(<FromBody()> ByVal value As String)

        Try
            Dim Location As DAL.Location = jss.Deserialize(value, GetType(DAL.Location))
            Location.Date = Date.Now

            If Not Business.Location.EntryExistsForToday(Location) Then Business.Location.Insert(Location)

        Catch ex As Exception
            Return Request.CreateErrorResponse(HttpStatusCode.Conflict, ex.Message)
        End Try

        Return "Location Logged"

    End Function

    ' PUT api/<controller>/5
    Public Sub PutValue(ByVal id As Integer, <FromBody()> ByVal value As String)

    End Sub

    ' DELETE api/<controller>/5
    Public Sub DeleteValue(ByVal id As Integer)

    End Sub
End Class
